### PR TITLE
refactor : Registered error handlers and changed return statements to raise

### LINKED ISSUE #6665: Register error handler for custom errors and change return statements to raise
https://github.com/fossasia/open-event-server/blob/9ba29f50ff2907e911dba2a51661015183ca658a/app/api/helpers/errors.py#L62-L75

https://flask.palletsprojects.com/en/1.1.x/errorhandling/#generic-exception-handlers

### PR DESCRIPTION
**Pull request by Rits1272**
Fixes #6665 
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/event_invoices.py [hunk 1]
```diff
@@ -171,7 +171,7 @@ def charge_paypal_payment_invoice(invoice_identifier):
         paypal_payment_id = request.json['data']['attributes']['paypal_payment_id']
         paypal_payer_id = request.json['data']['attributes']['paypal_payer_id']
     except Exception as e:
-        return BadRequestError({'source': e}, 'Bad Request Error').respond()
+        raise BadRequestError({'source': e}, 'Bad Request Error').respond()
```
Comment (iamareebjamal) [inline app/api/event_invoices.py]: No need of respond function. Have you even tested this?

File: app/instance.py [hunk 2]
```diff
@@ -261,13 +261,49 @@ def internal_server_error(error):
                          {'Content-Type': 'application/vnd.api+json'})
 
 
-@app.errorhandler(429)
```
Comment (mrsaicharan1) [inline app/instance.py]: Why are you replacing 429 with 422?

File: app/api/helpers/permission_manager.py [hunk 3]
```diff
@@ -483,11 +483,11 @@ def permission_manager(view, view_args, view_kwargs, *args, **kwargs):
             elif 'fetch' in kwargs:
                 kwargs[kwargs['fetch']] = fetched
         else:
-            return NotFoundError({'source': ''}, 'Object not found.').respond()
+            raise NotFoundError({'source': ''}, 'Object not found.')
     if args[0] in permissions:
         return permissions[args[0]](view, view_args, view_kwargs, *args, **kwargs)
     else:
-        return ForbiddenError({'source': ''}, 'Access forbidden').respond()
+        raise ForbiddenError({'source': ''}, 'Access forbidden')
```
Comment (niranjan94) [inline app/api/helpers/permission_manager.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 55
           

Clones removed
==============
+ app/api/orders.py  -1
+ app/api/role_invites.py  -2
+ app/api/event_invoices.py  -1
+ app/api/helpers/permission_manager.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277&bid=16053005)

File: app/api/helpers/permission_manager.py [hunk 4]
```diff
@@ -123,7 +123,7 @@ def is_user_itself(view, view_args, view_kwargs, *args, **kwargs):
     """
     user = current_user
     if not user.is_admin and not user.is_super_admin and user.id != kwargs['user_id']:
-        return ForbiddenError({'source': ''}, 'Access Forbidden').respond()
+        raise ForbiddenError({'source': ''}, 'Access Forbidden')
```
Comment (niranjan94) [inline app/api/helpers/permission_manager.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/helpers/permission_manager.py [hunk 5]
```diff
@@ -240,7 +240,7 @@ def is_registrar(view, view_args, view_kwargs, *args, **kwargs):
         return view(*view_args, **view_kwargs)
     if user.is_registrar(event_id) or user.has_event_access(event_id):
         return view(*view_args, **view_kwargs)
-    return ForbiddenError({'source': ''}, 'Registrar Access is Required.').respond()
+    raise ForbiddenError({'source': ''}, 'Registrar Access is Required.')
```
Comment (niranjan94) [inline app/api/helpers/permission_manager.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/helpers/errors.py [hunk 6]
```diff
@@ -42,8 +42,8 @@ def to_dict(self):
                 'title': self.title,
                 'detail': self.detail}
 
-
-class ForbiddenError(ErrorResponse):
+                
+class ForbiddenError(ErrorResponse):
```
Comment (niranjan94) [inline app/api/helpers/errors.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/helpers/permissions.py [hunk 7]
```diff
@@ -61,7 +61,7 @@ def is_super_admin(f):
     def decorated_function(*args, **kwargs):
         user = current_user
         if not user.is_super_admin:
-            return ForbiddenError({'source': ''}, 'Super admin access is required').respond()
+            raise ForbiddenError({'source': ''}, 'Super admin access is required')
```
Comment (niranjan94) [inline app/api/helpers/permissions.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/auth.py [hunk 8]
```diff
@@ -294,7 +294,7 @@ def reset_password_post():
     try:
         email = request.json['data']['email']
     except TypeError:
-        return BadRequestError({'source': ''}, 'Bad Request Error').respond()
+        raise BadRequestError({'source': ''}, 'Bad Request Error')
```
Comment (niranjan94) [inline app/api/auth.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/helpers/permission_manager.py [hunk 9]
```diff
@@ -389,37 +389,37 @@ def permission_manager(view, view_args, view_kwargs, *args, **kwargs):
...
-            return ForbiddenError({'source': ''}, 'Access forbidden').respond()
+            return ForbiddenError({'source': ''}, 'Access forbidden')
 
     # For Orders API
     if 'order_identifier' in view_kwargs:
         try:
             order = Order.query.filter_by(identifier=view_kwargs['order_identifier']).one()
         except NoResultFound:
-            return NotFoundError({'parameter': 'order_identifier'}, 'Order not found').respond()
+            raise NotFoundError({'parameter': 'order_identifier'}, 'Order not found')
```
Comment (niranjan94) [inline app/api/helpers/permission_manager.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/auth.py [hunk 10]
```diff
@@ -342,23 +342,23 @@ def change_password():
...
+                                       'Password should have minimum 8 characters')
             user.password = new_password
             save_to_db(user)
             send_email_with_action(user, PASSWORD_CHANGE,
                                    app_name=get_settings()['app_name'])
             send_notification_with_action(user, PASSWORD_CHANGE_NOTIF,
                                           app_name=get_settings()['app_name'])
         else:
-            return BadRequestError({'source': ''}, 'Wrong Password. Please enter correct current password.').respond()
+            raise BadRequestError({'source': ''}, 'Wrong Password. Please enter correct current password.')
```
Comment (niranjan94) [inline app/api/auth.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/helpers/permission_manager.py [hunk 11]
```diff
@@ -63,7 +63,7 @@ def is_organizer(view, view_args, view_kwargs, *args, **kwargs):
     if user.is_owner(kwargs['event_id']) or user.is_organizer(kwargs['event_id']):
         return view(*view_args, **view_kwargs)
 
-    return ForbiddenError({'source': ''}, 'Organizer access is required').respond()
+    raise ForbiddenError({'source': ''}, 'Organizer access is required')
```
Comment (niranjan94) [inline app/api/helpers/permission_manager.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/helpers/permissions.py [hunk 12]
```diff
@@ -232,7 +232,7 @@ def decorated_function(*args, **kwargs):
                     user.is_track_organizer(kwargs['event_id']) or
                     user.has_event_access(kwargs['event_id'])):
             return f(*args, **kwargs)
-        return ForbiddenError({'source': ''}, 'Track Organizer access is Required.').respond()
+        raise ForbiddenError({'source': ''}, 'Track Organizer access is Required.')
```
Comment (niranjan94) [inline app/api/helpers/permissions.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/helpers/permission_manager.py [hunk 13]
```diff
@@ -198,7 +198,7 @@ def is_speaker_itself_or_admin(view, view_args, view_kwargs, *args, **kwargs):
         if query_user:
             return view(*view_args, **view_kwargs)
 
-    return ForbiddenError({'source': ''}, 'Detail ownership is required, access denied.').respond()
+    raise ForbiddenError({'source': ''}, 'Detail ownership is required, access denied.')
```
Comment (niranjan94) [inline app/api/helpers/permission_manager.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/helpers/permission_manager.py [hunk 14]
```diff
@@ -317,7 +317,7 @@ def create_event(view, view_args, view_kwargs, *args, **kwargs):
         user = current_user
 
         if user.can_create_event is False:
-            return ForbiddenError({'source': ''}, 'Please verify your email').respond()
+            raise ForbiddenError({'source': ''}, 'Please verify your email')
```
Comment (niranjan94) [inline app/api/helpers/permission_manager.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/helpers/order.py [hunk 15]
```diff
@@ -138,7 +138,7 @@ def calculate_order_amount(tickets, discount_code):
     from app.api.helpers.ticketing import TicketingManager
     if discount_code:
         if not TicketingManager.match_discount_quantity(discount_code, tickets, None):
-            return UnprocessableEntityError({'source': 'discount-code'}, 'Discount Usage Exceeded').respond()
+            raise UnprocessableEntityError({'source': 'discount-code'}, 'Discount Usage Exceeded')
```
Comment (niranjan94) [inline app/api/helpers/order.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/helpers/permission_manager.py [hunk 16]
```diff
@@ -217,15 +217,15 @@ def is_session_self_submitted(view, view_args, view_kwargs, *args, **kwargs):
     try:
         session = Session.query.filter(Session.id == kwargs['session_id']).one()
     except NoResultFound:
-        return NotFoundError({'parameter': 'session_id'}, 'Session not found.').respond()
+        raise NotFoundError({'parameter': 'session_id'}, 'Session not found.')
```
Comment (niranjan94) [inline app/api/helpers/permission_manager.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/helpers/errors.py [hunk 17]
```diff
@@ -42,8 +42,8 @@ def to_dict(self):
                 'title': self.title,
                 'detail': self.detail}
 
-
-class ForbiddenError(ErrorResponse):
+
```
Comment (niranjan94) [inline app/api/helpers/errors.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/role_invites.py [hunk 18]
```diff
@@ -165,16 +165,16 @@ def accept_invite():
...
         try:
             user = User.query.filter_by(email=role_invite.email).first()
         except NoResultFound:
-            return NotFoundError({'source': ''}, 'User corresponding to role invite not Found').respond()
+            raise NotFoundError({'source': ''}, 'User corresponding to role invite not Found')
         try:
             role = Role.query.filter_by(name=role_invite.role_name).first()
         except NoResultFound:
-            return NotFoundError({'source': ''}, 'Role corresponding to role invite not Found').respond()
+            raise NotFoundError({'source': ''}, 'Role corresponding to role invite not Found')
```
Comment (niranjan94) [inline app/api/role_invites.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/event_invoices.py [hunk 19]
```diff
@@ -171,7 +171,7 @@ def charge_paypal_payment_invoice(invoice_identifier):
         paypal_payment_id = request.json['data']['attributes']['paypal_payment_id']
         paypal_payer_id = request.json['data']['attributes']['paypal_payer_id']
     except Exception as e:
-        return BadRequestError({'source': e}, 'Bad Request Error').respond()
+        raise BadRequestError({'source': e}, 'Bad Request Error')
```
Comment (niranjan94) [inline app/api/event_invoices.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/helpers/permissions.py [hunk 20]
```diff
@@ -98,7 +98,7 @@ def is_user_itself(f):
     def decorated_function(*args, **kwargs):
         user = current_user
         if not user.is_admin and not user.is_super_admin and user.id != kwargs['id']:
-            return ForbiddenError({'source': ''}, 'Access Forbidden').respond()
+            raise ForbiddenError({'source': ''}, 'Access Forbidden')
```
Comment (niranjan94) [inline app/api/helpers/permissions.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/helpers/permission_manager.py [hunk 21]
```diff
@@ -217,15 +217,15 @@ def is_session_self_submitted(view, view_args, view_kwargs, *args, **kwargs):
...
 
     if user.has_event_access(session.event_id):
         return view(*view_args, **view_kwargs)
 
     if session.creator_id == user.id:
         return view(*view_args, **view_kwargs)
 
-    return ForbiddenError({'source': ''}, 'Access denied.').respond()
+    raise ForbiddenError({'source': ''}, 'Access denied.')
```
Comment (niranjan94) [inline app/api/helpers/permission_manager.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/helpers/permission_manager.py [hunk 22]
```diff
@@ -275,7 +275,7 @@ def is_track_organizer(view, view_args, view_kwargs, *args, **kwargs):
         return view(*view_args, **view_kwargs)
     if user.is_track_organizer(event_id) or user.has_event_access(event_id):
         return view(*view_args, **view_kwargs)
-    return ForbiddenError({'source': ''}, 'Track Organizer access is Required.').respond()
+    raise ForbiddenError({'source': ''}, 'Track Organizer access is Required.')
```
Comment (niranjan94) [inline app/api/helpers/permission_manager.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/custom/invoices.py [hunk 23]
```diff
@@ -44,7 +44,7 @@ def order_invoices(order_identifier):
         try:
             order = Order.query.filter_by(identifier=order_identifier).first()
         except NoResultFound:
-            return NotFoundError({'source': ''}, 'Order Invoice not found').respond()
+            raise NotFoundError({'source': ''}, 'Order Invoice not found')
```
Comment (niranjan94) [inline app/api/custom/invoices.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/helpers/permissions.py [hunk 24]
```diff
@@ -120,7 +120,7 @@ def decorated_function(*args, **kwargs):
             return f(*args, **kwargs)
         if 'event_id' in kwargs and user.is_owner(kwargs['event_id']):
             return f(*args, **kwargs)
-        return ForbiddenError({'source': ''}, 'Owner access is required').respond()
+        raise ForbiddenError({'source': ''}, 'Owner access is required')
```
Comment (niranjan94) [inline app/api/helpers/permissions.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/helpers/permission_manager.py [hunk 25]
```diff
@@ -483,11 +483,11 @@ def permission_manager(view, view_args, view_kwargs, *args, **kwargs):
             elif 'fetch' in kwargs:
                 kwargs[kwargs['fetch']] = fetched
         else:
-            return NotFoundError({'source': ''}, 'Object not found.').respond()
+            raise NotFoundError({'source': ''}, 'Object not found.')
```
Comment (niranjan94) [inline app/api/helpers/permission_manager.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/helpers/permission_manager.py [hunk 26]
```diff
@@ -260,7 +260,7 @@ def is_registrar_or_user_itself(view, view_args, view_kwargs, *args, **kwargs):
     if user.is_registrar(event_id) or user.has_event_access(event_id):
         return view(*view_args, **view_kwargs)
 
-    return ForbiddenError({'source': ''}, 'Registrar access is required.').respond()
+    raise ForbiddenError({'source': ''}, 'Registrar access is required.')
```
Comment (niranjan94) [inline app/api/helpers/permission_manager.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

File: app/api/event_invoices.py [hunk 27]
```diff
@@ -149,7 +149,7 @@ def create_paypal_payment_invoice(invoice_identifier):
         return_url = request.json['data']['attributes']['return-url']
         cancel_url = request.json['data']['attributes']['cancel-url']
     except TypeError:
-        return BadRequestError({'source': ''}, 'Bad Request Error').respond()
+        raise BadRequestError({'source': ''}, 'Bad Request Error')
```
Comment (niranjan94) [inline app/api/event_invoices.py]: Codacy Issue found: [Raising a new style class which doesn't inherit from BaseException](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4832277)

**Pull request closed without merging**